src/contract.rs

```diff
@@ -858,11 +858,17 @@ pub fn execute_pass_through_swap(
         )?)
     };
 
+    let resp: InfoResponse = deps
+        .querier
+        .query_wasm_smart(&output_amm_address, &QueryMsg::Info {})?;
+    let transfer_input_token_enum = if transfer_token.denom == resp.token1_denom {
+        TokenSelect::Token1
+    } else {
```

Comment:
let's check `if transfer_token.denom == resp.token2_denom` too and otherwise throw an error

Reply:
done, made it throw and error if the output pool is invalid as well as extended the test to include this case.